### PR TITLE
GEODE-5971: Refactor AlterAsycnEventQueueCommand to use SingleGfshCommand

### PR DESCRIPTION
**Pull request by aditya87**
Co-authored-by: Jens Deppe <jdeppe@pivotal.io>
Co-authored-by: Kenneth Howe <khowe@pivotal.io>
Co-authored-by: Aditya Anchuri <aanchuri@pivotal.io>

Thank you for submitting a contribution to Apache Geode.

In order to streamline the review of the contribution we ask you
to ensure the following steps have been taken:

### For all changes:
- [X] Is there a JIRA ticket associated with this PR? Is it referenced in the commit message?

- [X] Has your PR been rebased against the latest commit within the target branch (typically `develop`)?

- [X] Is your initial contribution a single, squashed commit?

- [X] Does `gradlew build` run cleanly?

- [X] Have you written or updated unit tests to verify your changes?

- [ ] If adding new dependencies to the code, are these dependencies licensed in a way that is compatible for inclusion under [ASF 2.0](http://www.apache.org/legal/resolved.html#category-a)?

### Note:
Please ensure that once the PR is submitted, you check travis-ci for build issues and
submit an update to your PR as soon as possible. If you need help, please send an
email to dev@geode.apache.org.

### CODE REVIEW

File: geode-core/src/main/java/org/apache/geode/management/internal/cli/remote/CommandExecutor.java [hunk 1]
```diff
@@ -131,16 +136,31 @@ protected Object invokeCommand(Object command, GfshParseResult parseResult) {
     }
 
     String groupInput = parseResult.getParamValueAsString("group");
+    String[] groups;
     if (groupInput == null) {
-      groupInput = "cluster";
+      groups = new String[] {"cluster"};
```
Comment (jdeppe-pivotal) [inline geode-core/src/main/java/org/apache/geode/management/internal/cli/remote/CommandExecutor.java]: I like the overall general approach, however I don't like that the code flow here does not make it clear that `updateCacheConfig` and `updateConfigForGroup` are mutually exclusive.

**Comment by jinmeiliao** [comment]
How do you differentiate which interface method to call? in what situation we should update just one config object and in what situation we should go through all of them? I think this is determined by the nature of the command itself. Like "create aeq" needs to operate on only one group, while "delete aeq" needs to go through all groups and find the right one to delete. 

So the command needs to somehow indicate what kind of operation this is. We had suggested using a marker interface. If a command doesn't have that marker interface, then it's just doing what it is now, otherwise, the command executor will get all the configs and call the updateConfig method with each group's config. The interface of SingleGfshCommand doesn't need to change.

**Comment by aditya87** [comment]
@jinmeiliao Sure, I agree in general terms that we should set some kind of marker interface -- there's an interesting case where the destroy aeq command can run on a single config or can search through all configs, depending on whether the user provided a group option. We'll have to flush out how that could work.

**Comment by pdxrunner** [comment]
Removed the 2nd (confusing) `updateConfig` method from `SingleGfshCommand` in favor of using a marker interface as has been suggested.

**Comment by jinmeiliao** [comment]
@aditya87 that is an interesting case, I think in general, when a group is provided in the command, then we should just pass that config to the interface method. The marker is used to differentiate only  when there is no group supplied in the command, that determines should it be default to "cluster" group or apply to all groups.

**Comment by pdxrunner** [comment]
@jinmeiliao @aditya87 Good point. Our current implementation will always use the _all groups_ list for updating the configuration, not just those specified with a `--group` option.

File: geode-core/src/main/java/org/apache/geode/management/internal/cli/remote/CommandExecutor.java (hunk 1 above)
Comment (aditya87) [reply geode-core/src/main/java/org/apache/geode/management/internal/cli/remote/CommandExecutor.java]: Looks good with the new marker interface.

**Comment by aditya87** [comment]
Like the marker interface change 👍

File: geode-core/src/main/java/org/apache/geode/management/cli/UpdateAllConfigurationGroups.java [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+
+/**
+ * This interface is implemented by gfsh commands that can potentially update the configuration for
+ * all groups (including the cluster-wide group, "cluster").
+ * </p>
+ * CommandExecutor will builds the list of groups for configuration updating to include all groups
+ * if the command implements UpdateAllConfigurationGroups. Otherwise, the list of groups are those
+ * specified with a command --group option, or "cluster" if there is no --group option.
+ */
+public interface UpdateAllConfigurationGroups {
```
[the remark this answers is not in the document]
Comment (pdxrunner) [reply geode-core/src/main/java/org/apache/geode/management/cli/UpdateAllConfigurationGroups.java]: Good suggestion!

File: geode-core/src/main/java/org/apache/geode/management/internal/cli/remote/CommandExecutor.java [hunk 3]
```diff
@@ -130,17 +142,43 @@ protected Object invokeCommand(Object command, GfshParseResult parseResult) {
       return resultModel;
     }
 
+    String groupsToUpdate = "";
     String groupInput = parseResult.getParamValueAsString("group");
-    if (groupInput == null) {
-      groupInput = "cluster";
+    TabularResultModel table = null;
+    if (StringUtils.isBlank(groupInput)
```
Comment (jinmeiliao) [inline geode-core/src/main/java/org/apache/geode/management/internal/cli/remote/CommandExecutor.java]: For better readability:
if(!StringUtils.isBlank){
// use the group
}
else if (command is instance of UpdateAllMaker){
// pass all groups
}
else{
// pass in "cluster"
}

**Pull request merged**